### DISCUSSION TITLE
cannot use interface and timeout in library

### DISCUSSION THREAD

**Discussion by ArcherSking10** [thombashi/pingparsing#44] [Q&A] [answered]
when I wish to use pingparsing as library in my python code.
after I init a transmitter instance, and transmitter.timeout = 2, transmitter.interface = "eno1", actually the eno1 interface locally is DOWN, so the result should be null, but it returns the result go out from another interface instead of eno1

**Comment by thombashi** [comment, marked as answer]
Thank you for your report.

The problem had fixed at `pingparsing 1.3.1`
Please try to upgrade the package.
